libnavui-dropin/api/current.txt

```diff
@@ -28,6 +28,38 @@ package com.mapbox.navigation.dropin {
     property public final int style;
   }
 
+  public final class MapboxMapCompassParams {
+    method public boolean getClickable();
+    method public boolean getEnabled();
+    property public final boolean clickable;
+    property public final boolean enabled;
+  }
+
+  public static final class MapboxMapCompassParams.Builder {
+    ctor public MapboxMapCompassParams.Builder();
+    method public com.mapbox.navigation.dropin.MapboxMapCompassParams build();
+    method public com.mapbox.navigation.dropin.MapboxMapCompassParams.Builder clickable(boolean clickable);
+    method public com.mapbox.navigation.dropin.MapboxMapCompassParams.Builder enabled(boolean enabled);
+  }
+
+  public final class MapboxMapScalebarParams {
+    method public boolean getEnabled();
+    method public float getTextBorderWidth();
+    method public boolean isMetricUnits();
+    method public void setMetricUnits(boolean);
+    property public final boolean enabled;
+    property public final boolean isMetricUnits;
+    property public final float textBorderWidth;
+  }
+
+  public static final class MapboxMapScalebarParams.Builder {
+    ctor public MapboxMapScalebarParams.Builder(android.content.Context context);
+    method public com.mapbox.navigation.dropin.MapboxMapScalebarParams build();
+    method public com.mapbox.navigation.dropin.MapboxMapScalebarParams.Builder enabled(boolean enabled);
+    method public com.mapbox.navigation.dropin.MapboxMapScalebarParams.Builder isMetricsUnits(Boolean? isMetricUnits);
+    method public com.mapbox.navigation.dropin.MapboxMapScalebarParams.Builder textBorderWidth(float textBorderWidth);
+  }
```

Comment:
Do we really need any parameters other than `enabled`? Users can change them at any time using `mapView.compass.updateSettings` and `mapView.scalebar.updateSettings` and `MapView` reference is available via `registerMapObserver`.

Reply:
Yes, but this way we specify which parameters are supported. For example, if the user sets custom margins or position, it will work very poorly. But the things listed in  `MapboxMapScalebarParams` can be safely changed.

Reply:
There are a lot of other parameters that are definitely safe to change (e.g. colors), I don't think we should duplicate all of them. In my opinion it should be enough to list the parameters that are not supported in the documentation of `enabled` parameter.

Reply:
There is also the following concern. If we set a parameter and app sets the same parameter with different value, the last one wins. It might not be obvious to users and might cause them problems. What if we do like this: we only expose through our API those params that we set and that can be safely changed by a user? For now they are:
compass: enabled, image
scalebar: enabled, textBorderWidth, isMetricUnits

cc @abhishek1508

Reply:
I think we shouldn't have our own API for setting `textBorderWidth` as well. Since it is an issue on Maps SDK side, we shouldn't workaround it on Nav SDK side and certainly we don't want to add any APIs because of that, because we won't be able to remove them in the future.

Reply:
>I think we shouldn't have our own API for setting textBorderWidth as well. Since it is an issue on Maps SDK side, we shouldn't workaround it on Nav SDK side and certainly we don't want to add any APIs because of that, as we won't be able to remove them in the future.

I agree with this @dzinad 
This is an issue in the Maps SDK and we already have an issue logged. We shouldn't introduce workarounds unless the functionality is unusable. This problem is merely visible to the end user.

Reply:
>What if we do like this: we only expose through our API those params that we set and that can be safely changed by a user?

Thinking it through again, how is it going to help? Let's say we don't expose these params though our API. End users could still access them directly. Similarly, for the other properties, no matter we expose them or not, end users can still use them. This brings me back to me first [comment](https://github.com/mapbox/mapbox-navigation-android/pull/6355#discussion_r974157656). This is similar to what @Zayankovsky is saying [here](https://github.com/mapbox/mapbox-navigation-android/pull/6355#discussion_r977436272), but his argument is much simpler. 

Let's say, we don't expose `image` for `compass`, end users can still assign a different image to the compass.

Reply:
> We shouldn't introduce workarounds unless the functionality is unusable.

Not sure I agree. If we don't want to introduce workarounds, we should not expose the functionality at all until it's working properly on the Maps SDK side. If we do, it looks like: "We provide you with a Nav SDK, we depend on the Maps SDK, this thing does not work properly but that's not our problem". 

> and certainly we don't want to add any APIs because of that, as we won't be able to remove them in the future.

That's a good point, let's remove it then and hardcode 3f. So the default behaviour is fine, it the user changes it themselves, they will probably notice that something's off, then, if they come to us, we'll say that it will be fixed in the near future, but for now we support the default border (or a smaller one). BTW, we can specify it in the docs. We can remove the docs.

Reply:
>Not sure I agree. If we don't want to introduce workarounds, we should not expose the functionality at all until it's working properly on the Maps SDK side. If we do, it looks like: "We provide you with a Nav SDK, we depend on the Maps SDK, this thing does not work properly but that's not our problem".

I don't mean we should do this always. In this case the workaround is introducing APIs that we cannot remove in the future unless the next major release.

Reply:
I suggested not adding the API in this case.